kyo-prelude/shared/src/main/scala/kyo/Abort.scala

```diff
@@ -21,12 +21,12 @@ object Abort:
 
     inline def panic[E](inline ex: Throwable)(using inline frame: Frame): Nothing < Abort[E] = error(Panic(ex))
 
-    private[kyo] inline def error[E](inline e: Error[E])(using inline frame: Frame): Nothing < Abort[E] =
+    inline def error[E](inline e: Error[E])(using inline frame: Frame): Nothing < Abort[E] =
         ArrowEffect.suspendMap[Any](erasedTag[E], e)(_ => ???)
 
-    inline def when[E](b: Boolean)(inline value: => E)(using inline frame: Frame): Unit < Abort[E] =
-        if b then fail(value)
-        else ()
+    inline def ensuring[A, E](cond: Boolean, result: => A = ())(inline value: => E)(using inline frame: Frame): A < Abort[E] =
```

Comment:
Not sure I like this change (at my own suggestion). I think it may be confusing to have default unit.

Reply:
maybe we could keep both `ensuring` and `when` to avoid the unit default? The new `ensuring` method seems quite useful

Reply:
That's fine with me!